src/scanner.c

```diff
@@ -417,18 +444,30 @@ static inline bool scan_close_bracket2(TSLexer* lexer, Stack* stack) {
 }
 
 static bool scan(TSLexer* lexer, Stack* stack, const bool* valid_symbols) {
+  if (valid_symbols[ERROR_SENTINEL]) {
+    // Decline to handle when in "error recovery" mode. When a syntax error occurs,
+    // tree-sitter calls the external scanner with all `valid_symbols` marked as valid.
+    return false;
+  }
```

Comment:
Important to remember this weird "error recovery" mode where _every_ symbol gets marked as valid (so our `START` symbol would incorrectly get marked as valid a 2nd time around if we aren't careful)